src/entity/UserPost.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  Index,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post } from './posts';
+import { User } from './User';
+
+export type UserPostFlags = Partial<{
+  feedbackDismiss: boolean;
+}>;
+
+export type UserPostFlagsPublic = Pick<UserPostFlags, 'feedbackDismiss'>;
+
+export enum UserPostVote {
+  Up = 1,
+  None = 0,
+  Down = -1,
+}
+
+export const userPostDefaultData = Object.freeze({
+  vote: UserPostVote.None,
+  hidden: false,
+  flags: {},
+});
+
+@Entity()
+@Index(['postId', 'userId'], { unique: true })
+export class UserPost {
+  @PrimaryColumn({ type: 'text' })
+  postId: string;
+
+  @PrimaryColumn({ type: 'text' })
+  userId: string;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column({ type: 'smallint', default: UserPostVote.None })
```

Comment:
Can we use tinyint perhaps? Much smaller

```suggestion
  @Column({ type: 'tinyint', default: UserPostVote.None })
```

Reply:
`tinyint` is [not supported in psql ](https://www.postgresql.org/docs/8.1/datatype.html) from what I know. `smallint` should be closest alternative.

Reply:
Ok!